odo/backends/sql.py

```diff
@@ -321,7 +320,7 @@ def select_to_base(sel, dshape=None, **kwargs):
     engine = sel.bind  # TODO: get engine from select
 
     with engine.connect() as conn:
-        result = conn.execute(sel)
+        result = conn.execute(sel).fetchall()
```

Comment:
Batching sounds good.  I'm surprised that sqlalchemy doesn't do this
already.

On Mon, Mar 30, 2015 at 7:16 AM, Phillip Cloud <notifications@github.com>
wrote:

> In odo/backends/sql.py
> <https://github.com/ContinuumIO/odo/pull/155#discussion_r27393299>:
>
> > @@ -321,7 +320,7 @@ def select_to_base(sel, dshape=None, **kwargs):
> >      engine = sel.bind  # TODO: get engine from select
> >
> >      with engine.connect() as conn:
> > -        result = conn.execute(sel)
> > +        result = conn.execute(sel).fetchall()
>
> Hm. Laziness *does* have the benefit of not blowing up memory, but
> without fetchall sqlalchemy can and often does hit the server with every
> call to fetchone. This of course depends on the dialect. I like laziness
> but hitting the server every single time for, say, a 1000 row result set is
> crazy. A happy medium might be to do something along the lines of:
>
> result = conn.execute(sel)
> chunksize = 1000
> def execute(conn, sel, chunksize=1000):
>     result = conn.execute(sel)
>     while True:
>         try:
>             yield result.fetchmany(size=chunksize)
>         except StopIteration:
>             return
> for row in concat(execute(conn, sel)):
>     yield row
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ContinuumIO/odo/pull/155/files#r27393299>.
>